### ISSUE TITLE
Stuck by creating new model for linear1D

### ISSUE THREAD

**Issue by Arsmart123** [hpcaitech/ColossalAI#136]
### Describe the feature

So I am trying to run a new model. On my local PC, I try to run 1d parallel with TENSOR_PARALLEL_SIZE=1, because my PC has only one GPU, and this model works. But on HPC, when I try TENSOR_PARALLEL_SIZE=2, (only 2 GPUs), this model is blocked and does not move! Any suggestions? Thank you!!!

My command to run this code is 

DATA='./dataset/' torchrun --nproc_per_node='2'  --nnodes='1'   --node_rank='0'  --master_addr='172.18.126.98'  --master_port='51063'   train_1.py --config='./configs/vit_1d.py'

and I have attached the MobaXterm screen below. 
//////////////////////////////////////////////////////////////////////////
'''
import os
import colossalai
import torchvision
from colossalai.builder import *
from colossalai.core import global_context as gpc
from colossalai.logging import get_dist_logger
from colossalai.nn import Accuracy, CrossEntropyLoss, MSELoss
from colossalai.nn.lr_scheduler import CosineAnnealingWarmupLR
from colossalai.trainer import Trainer
from colossalai.trainer.hooks import (AccuracyHook, LogMemoryByEpochHook,
                                      LogMetricByEpochHook,
                                      LogMetricByStepHook,
                                      LogTimingByEpochHook, LossHook,
                                      LRSchedulerHook, ThroughputHook)
from colossalai.utils import MultiTimer, get_dataloader
from model_zoo.vit import vit_lite_depth7_patch4_32
from torchvision import transforms
from colossalai.nn import Linear1D
import torch.nn as nn
from torch.utils.data import Dataset, DataLoader
import torch
os.environ['MASTER_ADDR'] = '172.18.126.98'
os.environ['MASTER_PORT'] = '51064'
os.environ['DATA'] = './dataset/'
os.environ['CONFIG_FILE'] = './configs/vit_1d.py'
os.environ['LOCAL_RANK'] = '0'
os.environ['RANK'] = '0'
os.environ['WORLD_SIZE'] = '2'
DATASET_PATH = str(os.environ['DATA'])

class trainset(Dataset):

    def __init__(self):
        pass

    def __getitem__(self, index):
        target = torch.randn(1)
        data = torch.randn(1, 1024)
        return data, target

    def __len__(self):
        return 512*5

def build_cifar(batch_size):

    train_dataset = trainset()
    test_dataset = trainset()
    train_dataloader = get_dataloader(dataset=train_dataset,
                                      shuffle=True,
                                      batch_size=batch_size,
                                      num_workers=0,
                                      pin_memory=True)
    test_dataloader = get_dataloader(dataset=test_dataset, batch_size=batch_size, num_workers=0, pin_memory=True)
    return train_dataloader, test_dataloader

class MLP_1D(nn.Module):

    def __init__(self):
        super().__init__()
        self.linear_1 = Linear1D(in_features=1024, out_features=16384)
        self.linear_2 = Linear1D(in_features=16384, out_features=1)

    def forward(self, x):
        x = self.linear_1(x)
        x = self.linear_2(x)
        x = x.squeeze(-1)
        return x

def train_cifar():
    args = colossalai.get_default_parser().parse_args()

    colossalai.launch_from_torch(config=args.config)
    print('111')
    logger = get_dist_logger()
    if hasattr(gpc.config, 'LOG_PATH'):
        if gpc.get_global_rank() == 0:
            log_path = gpc.config.LOG_PATH
            if not os.path.exists(log_path):
                os.mkdir(log_path)
            logger.log_to_file(log_path)
    print('1.5 1.5 1.5')
    model = MLP_1D()  # !!!!!!!!!Here is blocked!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

    print('222')
    train_dataloader, test_dataloader = build_cifar(gpc.config.BATCH_SIZE // gpc.data_parallel_size)

    criterion = CrossEntropyLoss(label_smoothing=0.1)

    optimizer = torch.optim.AdamW(model.parameters(), lr=gpc.config.LEARNING_RATE, weight_decay=gpc.config.WEIGHT_DECAY)
    print('333')
    steps_per_epoch = len(train_dataloader)

    lr_scheduler = CosineAnnealingWarmupLR(optimizer=optimizer,
                                           total_steps=gpc.config.NUM_EPOCHS * steps_per_epoch,
                                           warmup_steps=gpc.config.WARMUP_EPOCHS * steps_per_epoch)

    engine, train_dataloader, test_dataloader, lr_scheduler = colossalai.initialize(model=model,
                                                                                    optimizer=optimizer,
                                                                                    criterion=criterion,
                                                                                    train_dataloader=train_dataloader,
                                                                                    test_dataloader=test_dataloader,
                                                                                    lr_scheduler=lr_scheduler)

    logger.info("Engine is built", ranks=[0])

    timer = MultiTimer()
    print('444')
    trainer = Trainer(engine=engine, logger=logger, timer=timer)
    logger.info("Trainer is built", ranks=[0])

    hooks = [
        LogMetricByEpochHook(logger=logger),
        LogMetricByStepHook(),
        AccuracyHook(accuracy_func=Accuracy()),
        LossHook(),
        ThroughputHook(),
        LRSchedulerHook(lr_scheduler=lr_scheduler, by_epoch=False)
    ]
    print('555')
    logger.info("Train start", ranks=[0])
    trainer.fit(train_dataloader=train_dataloader,
                test_dataloader=test_dataloader,
                epochs=gpc.config.NUM_EPOCHS,
                hooks=hooks,
                display_progress=True,
                test_interval=1)

if __name__ == '__main__':
    train_cifar()
'''

![22222](https://user-images.githubusercontent.com/65449458/148644995-abf051da-ad00-46d5-869d-4986fc800baf.png)

**Comment by Arsmart123** [comment]
No need for furthur help, I solved it! These lines is useless, and just removing them is enough. 

os.environ['MASTER_ADDR'] = '172.18.126.98'
os.environ['MASTER_PORT'] = '51064'
os.environ['DATA'] = './dataset/'
os.environ['CONFIG_FILE'] = './configs/vit_1d.py'
os.environ['LOCAL_RANK'] = '0'
os.environ['RANK'] = '0'
os.environ['WORLD_SIZE'] = '2'
DATASET_PATH = str(os.environ['DATA'])
